### PR TITLE
Add RzBaseType listing API

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

- Renamed recently added functions like `rz_type_db_enums()`, `rz_type_db_structs()` into `rz_type_db_enum_names()` since they return only names of the corresponding types for listing and autocompletion purposes
- Added proper listing API: `rz_type_db_get_base_types()` and `rz_type_db_get_base_types_kind()`
- Removed broken Visual Types feature (`Vbt` command)

Removed Visual Types produced the following output which is largely useless:

image

After the complete "Types" refactoring project is finished it makes sense to implement such a navigation properly.

**Test plan**

CI is green

### CODE REVIEW

File: librz/type/base.c [hunk 1]
```diff
@@ -277,6 +282,69 @@ RZ_API RzBaseType *rz_type_db_get_base_type(RzTypeDB *typedb, const char *name)
 	return base_type;
 }
 
+/**
+ * \brief Returns the list of all basic types of the chosen kind
+ *
+ * \param typedb Types Database instance
+ * \param kind Kind of the types to list
+ */
+RZ_API RZ_OWN RzList /* RzBaseType */ *rz_type_db_get_base_types_kind(RzTypeDB *typedb, RzBaseTypeKind kind) {
```
Comment (ret2libc) [inline librz/type/base.c]: Would it make sense to have a `ANY` kind and just use one API instead of two that do the same thing?

**Review by ret2libc — APPROVED** [review]

File: librz/type/base.c (hunk 1 above)
Comment (XVilka) [reply librz/type/base.c]: It wouldn't make sense to add in `RzBaseTypeKind`.

Comment (XVilka) [reply librz/type/base.c]: Also note, that the bodies of these functions will change when I migrate types storage to the hashtable.

Comment (thestr4ng3r) [reply librz/type/base.c]: agree with @XVilka 
I would call this function `rz_type_db_get_base_types_of_kind()` though, so it's a bit more descriptive.

**Review by thestr4ng3r — COMMENTED** [review]
overall lgtm

File: librz/type/base.c (hunk 1 above)
Comment (XVilka) [reply librz/type/base.c]: Renamed the function.

**Review by thestr4ng3r — APPROVED** [review]

**Pull request merged**